deepchecks/base/dataset.py

```diff
@@ -124,15 +134,16 @@ def __init__(
                 if label_name in self._data.columns:
                     raise DeepchecksValueError('Can\'t pass label with label_name that exists in the data. change '
                                                'the label_name parameter')
-            self._data[label_name] = label
+            # Cast to numpy to promise no index collisions
+            self._data[label_name] = np.array(label)
 
-        if use_default_index is True and index_name is not None:
-            raise DeepchecksValueError('parameter use_default_index cannot be True if index is given')
+        if create_index_from_dataframe_index is True and index_name is not None:
```

Comment:
Guessing here you just renamed the previous code as part of the refactor cause it's not yet the behavior describe above :)

Reply:
yep(: